File: worldeditsource/worldedit/WorldEditWorldData.scala
package jk_5.nailed.worldedit

import com.sk89q.worldedit.world.registry.LegacyWorldData

/**
 * No description given
 *
 * @author jk-5
 */
object WorldEditWorldData extends LegacyWorldData {
  override def getBiomeRegistry = WorldEditBiomeRegistry
}

File: src/main/java/jk_5/nailed/plugins/worldedit/WorldEditWorldData.java
package jk_5.nailed.plugins.worldedit;

import com.sk89q.worldedit.world.registry.BiomeRegistry;
import com.sk89q.worldedit.world.registry.LegacyWorldData;

/**
 * World data for the Forge platform.
 */
class WorldEditWorldData extends LegacyWorldData {

    private static final WorldEditWorldData INSTANCE = new WorldEditWorldData();
    private final BiomeRegistry biomeRegistry = new WorldEditBiomeRegistry();

    /**
     * Create a new instance.
     */
    WorldEditWorldData() {
    }

    @Override
    public BiomeRegistry getBiomeRegistry() {
        return biomeRegistry;
    }

    /**
     * Get a static instance.
     *
     * @return an instance
     */
    public static WorldEditWorldData instance() {
        return INSTANCE;
    }
}
